### PR TITLE
refactor(#1296): A1 - migrate LoadData/SaveData to explicit ADR-043 FormatCapability

### LINKED ISSUE #1296: track(#1204): migrate in-tree core IO + imaging + SRS packages to ADR-043 FormatCapability + add imaging formats + ProcessBlock ome propagation contract
## Parent / Context

Sub-issue of #1204 ("Track ADR-043 package migration to explicit IO format
capabilities"). Implements the manager-coordinated migration slice spanning
in-tree core IO blocks (LoadData/SaveData), the shipped scieasy-blocks-imaging
package, and the shipped scieasy-blocks-srs package. LCMS package migration is
explicitly out of scope and tracked separately under #1204.

## Implementation Spec

`docs/specs/adr-043-package-migration.md` (committed alongside this work).
Spec follows ADR-042 §3.4 SpecKit structure. FR-001..FR-017 + SC-001..SC-006.

## Scope

In-scope:

- LoadData / SaveData declare explicit FormatCapability records covering all 6
  core DataObject types × supported extensions; delete legacy
  `supported_extensions` ClassVar.
- imaging LoadImage / SaveImage declare explicit FormatCapability records.
- imaging adds PNG/JPEG (Pillow) + Bio-Formats microscopy family (CZI/ND2/LIF/
  OIR/OIB minimum) via `cellprofiler/python-bioformats`, load-only, shipped as
  the `imaging[bioformats]` optional install extra (cellpose pattern).
- `Image.Meta` and `Label.Meta` gain `ome: ome_types.model.OME | None` typed
  field. `ome-types>=0.5,<0.6` becomes a required dep of scieasy-blocks-imaging.
- ProcessBlock OME metadata propagation contract codified (Modes A/B/C);
  imaging + SRS ProcessBlock audit + fix.
- Frontend: capability dropdown on port editor, OME metadata browser, lossy-save
  warning.
- Owner-authored end-to-end test phase (Phase D) before final merge.

Out of scope:

- LCMS package migration (separate sub-issue under #1204).
- OME-Zarr v0.4 first-class file-format support (vanilla zarr stays).
- Adding LoadSRSImage IO block (separate issue if desired).
- Bio-Formats save support (library is load-only by design).
- Engine / registry / validator / materialisation changes (already migrated).

## Implementation Plan

| Phase | Surface | Agents | Dependency |
|---|---|---|---|
| A1 | core IO (LoadData/SaveData) | 1 implementer | none |
| A2 | imaging IO + Image.Meta.ome + Bio-Formats extras | 1 implementer | none |
| A3 | frontend UI (dropdown / OME browser / lossy-save warning) | 1 implementer | none |
| B1 | imaging ProcessBlock propagation audit + fix | 1 implementer | A2 merged |
| B2 | SRS ProcessBlock propagation audit + fix | 1 implementer | A2 merged |
| C1 | no-context audit | 1 audit_reviewer | A+B merged |
| C2 | integration verification | manager | C1 done |
| D1 | owner-authored e2e test cases | owner | C2 done |
| D2 | execute owner e2e cases | implementer / manager | D1 done |
| D3 | e2e gate before final merge | manager | D2 all-pass |

Umbrella branch: `track/adr-043/core-blocks-and-imaging` off main.
Umbrella PR: `[DO NOT MERGE]` pointing at main; sub-PRs target umbrella.
Manager checklist: `docs/planning/adr-043-package-migration-checklist.md`.

## Acceptance Criteria

- FR-001..FR-017 implemented and verified per spec §4.4 verification plan.
- SC-001..SC-006 measurable outcomes green.
- ADR-043 §9 package validity scan passes on LoadData/SaveData/LoadImage/SaveImage.
- Bio-Formats family loads CZI/ND2/LIF/OIR fixtures with populated `Image.Meta.ome`.
- OME propagation golden path (CZI -> resize -> save TIFF) preserves pixel_size.
- Frontend Chrome smoke green.
- Every owner-authored e2e case in Phase D passes.
- All sub-PRs merged into umbrella; umbrella PR CI green; owner approves final PR.

## Related

- ADR-041: `docs/adr/ADR-041.md` (CodeBlock v2)
- ADR-043: `docs/adr/ADR-043.md` (IO Format Capability Registry)
- Spec: `docs/specs/adr-043-package-migration.md`
- Parent tracking: #1204

### PR DESCRIPTION
**Pull request by jiazhenz026**
## Summary

ADR-043 / spec `docs/specs/adr-043-package-migration.md` Phase A1 (FR-001 / FR-002 / FR-003 / FR-015 / FR-016).

This is the A1 (Core IO) sub-PR of the ADR-043 in-tree-and-imaging package migration umbrella ([#1297](https://github.com/zjzcpj/SciEasy/pull/1297) `[DO NOT MERGE]`). It targets the umbrella branch `track/adr-043/core-blocks-and-imaging`, NOT `main`.

- Declare `LoadData.format_capabilities` (30 records) and `SaveData.format_capabilities` (31 records) as explicit `ClassVar[tuple[FormatCapability, ...]]` covering all six core `DataObject` types x supported extensions per the FR-015 capability-id convention `core.{lower(type)}.{format_id}.{load|save}`. Every record is `is_synthesized=False`, `is_default=True`, `MetadataFidelity(level="pixel_only")`, and load+save siblings share `roundtrip_group=core.{type}.{format}`. Pickle records carry `notes='requires allow_pickle=True'`.

- Delete the legacy `supported_extensions: ClassVar[dict[str, str]]` from `LoadData` and `SaveData` class bodies per FR-003. Override `_detect_format` on each class to consult the capability-derived module-level dispatch map (`_LOAD_EXTENSION_MAP` / `_SAVE_EXTENSION_MAP`). Rewire `_resolve_format` / `_resolve_save_format` and the user-facing "Supported extensions" error messages through `_supported_load_extensions()` / `_supported_save_extensions()`.

- Add `Artifact`-as-opaque-loader / -saver capability records covering the typed-core extension union (`.csv` / `.parquet` / `.npy` / `.zarr` / `.text` extensions / `.pickle` / ...). These mirror the pre-ADR-043 synthesized `data_type=DataObject` capability that the AppBlock wildcard-port binner (`types=['DataObject']` mapped to `target_type=Artifact`) relied on, so the binner keeps resolving a loader/saver against the explicit capability set without falling through to the registry's polymorphic fallback (which is now legitimately ambiguous because per-concrete-type declarations no longer collapse to a single `data_type=DataObject` synthesized record).

- Add `tests/blocks/io/test_load_data_capabilities.py` (47 cases) and `tests/blocks/io/test_save_data_capabilities.py` (50 cases) covering declaration shape (FR-001/FR-002/FR-015), per-type coverage (FR-016), registry resolution via `find_loader_capability` / `find_saver_capability`, ambiguity error for two non-default candidates at the same slot, pickle gating, and backward-compat round-trip parity for DataFrame+CSV / Array+npy / Text+md.

- Rewrite the legacy `TestSupportedExtensionsClassVar` test classes in `tests/blocks/io/test_load_data.py` and `tests/blocks/io/test_save_data.py` to assert the capability-derived contract (the legacy ClassVar is gone). Update `tests/blocks/app/test_appblock_bin_outputs.py` to use the new explicit `core.dataframe.csv.load` capability id in place of the deleted synthesized id - direct FR-003 consequences (Tests are part of implementation work per AGENTS.md S3.1).

Closes #1296

## Test plan

- [x] `python -m ruff check src/scieasy/blocks/io/loaders/ src/scieasy/blocks/io/savers/ tests/blocks/io/ tests/blocks/app/test_appblock_bin_outputs.py` -> pass
- [x] `python -m ruff format --check src/scieasy/blocks/io/loaders/ src/scieasy/blocks/io/savers/ tests/blocks/io/ tests/blocks/app/test_appblock_bin_outputs.py` -> pass
- [x] `PYTHONPATH=src python -m pytest tests/blocks/io/ tests/blocks/app/test_appblock_bin_outputs.py 'tests/blocks/test_app_block.py::TestAppBlockExtensionBinner' --timeout=60 --no-cov` -> 281 passed
- [x] `python -m scieasy.qa.audit.full_audit --repo-root . --format json --output docs/audit/full-audit-latest.json` -> fail; 0/311 reported findings touch the A1 write set; remaining are pre-existing repo-wide ADR-031/architecture/signature/closure debt owner-acknowledged at umbrella level (manager track 5.: `admin-approved:ai-override`).
- [ ] Sentrux: `skipped` - Sentrux CLI unavailable in this worktree; tests + ruff + ruff format + full_audit cover the change surface. Same rationale as manager-track umbrella.
- [ ] CI to run after this PR opens.
- [ ] Codex auto-review reconciliation (cap at one round per ADR-042 norms).

## Gate record

`.workflow/records/1296-a1-core-io.json`

## Local-hook bypass evidence

`admin-approved:ai-override` (mirrors umbrella manager-track authorization, chat 2026-05-20):

- Pre-push hook: passes WITH `--bypass-label admin-approved:ai-override` because (a) Sentrux is `skipped` (CLI unavailable in worktree); (b) `commit_and_submit_pr` stage advances at `finalize` time (post-PR-open) per ADR-042 Addendum 1 §3.2. Without the bypass label the pre-push validator returns `gate-record.sentrux.not-passing` and `gate-record.stage.not-done: commit_and_submit_pr` - both expected pre-finalize.
- PR-create: `admin-approved:ai-override` label applied at `gh pr create` time per ADR-042 Addendum 1 §3.2 Step 5.

## Out of scope (deferred to other A1+ tracks)

- imaging IO migration (A2, sibling agent)
- frontend capability/UI scaffolding (A3, sibling agent)
- ProcessBlock OME propagation (B1/B2, sequenced after A2)
- LCMS package migration (separate sub-issue under #1204)
- `src/scieasy/blocks/io/io_block.py` - base class `supported_extensions` ClassVar stays (FR-003 explicitly scopes the delete to LoadData/SaveData only)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/scieasy/blocks/io/savers/save_data.py [hunk 1]
```diff
@@ -67,24 +76,398 @@
...
+        extensions=(
+            ".txt",
+            ".log",
+            ".md",
+            ".html",
+            ".xml",
+            ".yaml",
+            ".yml",
+            ".toml",
+        ),
```
[the remark this answers is not in the document]
Comment (jiazhenz026) [reply src/scieasy/blocks/io/savers/save_data.py:277]: **ACCEPTED.** Fixed in commit (incoming after this reply). Added two Text save capability records: `core.text.text.save` for the load-paired text extensions (`.txt/.log/.md/.html/.xml/.yaml/.yml/.toml`) and a separate `core.text.json.save` for the legacy save-only `.json` extension. The split is required because the legacy extension-map derivation rejects extension-to-format-id conflicts (one extension cannot map to two distinct format_ids), so `.json` -> "text" would collide with `.json` -> "json" from DataFrame/Series/CompositeData/Artifact. Updated tests/blocks/io/test_save_data_capabilities.py per-type coverage and roundtrip-group pairing to allow `core.text.json` as a save-only legacy edge case alongside the existing `core.series.json` exception. `find_saver_capability(Text, '.json')` now resolves to `core.text.json.save` instead of raising MissingCapabilityError.
